include/envoy/event/dispatcher.h

```diff
@@ -41,20 +41,26 @@ class Dispatcher {
   /**
    * Create a client connection.
    * @param address supplies the address to connect to.
+   * @source_address supplies an optional address to bind to.
    * @return Network::ClientConnectionPtr a client connection that is owned by the caller.
    */
   virtual Network::ClientConnectionPtr
-  createClientConnection(Network::Address::InstanceConstSharedPtr address) PURE;
+  createClientConnection(Network::Address::InstanceConstSharedPtr address,
+                         Optional<Network::Address::InstanceConstSharedPtr> source_address =
+                             Optional<Network::Address::InstanceConstSharedPtr>()) PURE;
 
   /**
    * Create an SSL client connection.
    * @param ssl_ctx supplies the SSL context to use.
    * @param address supplies the address to connect to.
+   * @source_address supplies an optional address to bind to.
    * @return Network::ClientConnectionPtr a client connection that is owned by the caller.
    */
   virtual Network::ClientConnectionPtr
   createSslClientConnection(Ssl::ClientContext& ssl_ctx,
-                            Network::Address::InstanceConstSharedPtr address) PURE;
+                            Network::Address::InstanceConstSharedPtr address,
+                            Optional<Network::Address::InstanceConstSharedPtr> source_address =
+                                Optional<Network::Address::InstanceConstSharedPtr>()) PURE;
```

Comment:
I think I'd prefer to just use the default constructed/null/empty shared pointer instead of adding Optional.